src/rust/engine/options/src/config.rs

```diff
@@ -217,21 +217,132 @@ fn toml_table_to_dict(table: &Value) -> HashMap<String, Val> {
 }
 
 #[derive(Clone)]
-pub(crate) struct Config {
+struct ConfigSource {
+    #[allow(dead_code)]
+    path: Option<OsString>,
```

Comment:
Why do we have this field if it isn't used anywhere? It seems weird that we go through the trouble of converting a path to an osttring to not use it.

Reply:
We'll be using it soon, to report the sources of the config to the user in help-all.